kolibri/plugins/facility/assets/src/routes.js

```diff
@@ -24,21 +24,45 @@ import {
 } from './modules/classAssignMembers/handlers';
 import { PageNames } from './constants';
 
-function facilityParamRequiredGuard(toRoute, subtopicName) {
-  if (store.getters.userIsMultiFacilityAdmin && !toRoute.params.facility_id) {
-    router.replace({
-      name: 'ALL_FACILITIES_PAGE',
-      params: { subtopicName },
-    });
-    return true;
+function facilityParamRequiredGuard(toRoute, fromRoute, subtopicName) {
+  if (
+    store.getters.userIsMultiFacilityAdmin &&
+    !toRoute.params.facility_id &&
+    fromRoute.query.subtopicName !== subtopicName
```

Comment:
`fromRoute.query.subTopicName` will always be undefined, so this will always be true.

I don't think you need to make this comparison here at all...